Added Spring Boot client for illustrating accessing a resource server

roar-skinderviken opened the pull request:
### Start the auth-server on port 9000
```shell
./gradlew -p auth-server bootRun
```

### Start the resource-server on port 8080
```shell
./gradlew -p backend-spring-boot bootRun
```

### Start the spring-boot-client on port 8085
```shell
./gradlew -p spring-boot-client bootRun
```

Navigate to
http://localhost:8085/index

Select Secured endpoints | Messages

You should now be redirected to http://localhost:9000/auth-server/login

Username/password: user1/password

OhadR commented:
looks great. but please explain: how do u trigger the oauth process? in the client app i see controller `/messages`, which i assume is called from the browser (?). if the user has not ligin yet - do you get a redirect to the auth-server login screen?

roar-skinderviken commented:
@OhadR 
> if the user has not ligin yet - do you get a redirect to the auth-server login screen?
Yes, that is correct.

OhadR commented:
i see. but in my case i do not use WebClient, but RestClient... does this matter?
by the way, i figured out that the magic happens via the controller of the client:
```
@GetMapping(value = "/messages")
    public String authorizationCodeGrant(
            Model model,
            @RegisteredOAuth2AuthorizedClient("messaging-client-oidc")
            OAuth2AuthorizedClient authorizedClient) {
```

but now the question is, when the client calls client domain, and being redirected to the auth-server, how come you dont get CORS error?

roar-skinderviken commented:
@OhadR Code for using RestClient is a little bit different. I added code for that in this PR: https://github.com/roar-skinderviken/vicx-applications/pull/161

I also added the code in a answer to your StackOverflow question: https://stackoverflow.com/a/79355075/14072498

CORS is not an issue when client browser is redirected.
